packages/@lwc/compiler/src/transformers/transformer.ts

```diff
@@ -17,6 +22,7 @@ import scriptTransformer from './javascript';
 export interface TransformResult {
     code: string;
     map: unknown;
+    warnings?: CompilerDiagnostic[];
```

Comment:
The style and babel compilers don't have warnings propagated here, although I suppose we could do that in the future.